### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by IliaRusecki**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,125 @@
...
+            </properties>
+        </profile>                             -->
+
+        <profile>
+            <id>hw2</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <suite.file>src/test/resources/HW2TestSuite.xml</suite.file>
```
Comment (khda91) [inline pom.xml]: it is better store suites files in the related folder for the homeworks in the `resources`

File: src/test/java/hw2/ex1/Ex1Tests.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        String userName = driver.findElement(By.id("user-name")).getText();
+        assertEquals(userName, "PITER CHAILOVSKII");
+
+
+        //5. ASSERT BROWSER TITLE
+        assertEquals(browserTitle, "Home Page");
+
+
+        //6.  ASSERT THAT THERE ARE 4 ITEMS ON THE HEADER SECTION ARE DISPLAYED AND THEY HAVE PROPER TEXTS
+        List<WebElement> items = driver.findElements(By.xpath("//ul[@class= 'uui-navigation nav navbar-nav m-l8']/li"));
```
Comment (khda91) [inline src/test/java/hw2/ex1/Ex1Tests.java]: Why do you decide list all classes for the `div`?

File: src/test/java/hw2/ex1/Ex1Tests.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        driver.switchTo().frame(iframe);
+        WebElement logo = driver.findElement(By.id("epam_logo"));
+        assertTrue(logo.isDisplayed());
+
+        //12. SWITCH TO ORIGINAL WINDOW BACK
+        driver.switchTo().window(mainPage);
+        assertEquals(driver.getTitle(), "Home Page");
+
+        //13. ASSERT A TEXT OF THE SUB HEADER
+        WebElement subHeader = driver.findElement(By.xpath("//a[text() = 'JDI Github']"));
```
Comment (khda91) [inline src/test/java/hw2/ex1/Ex1Tests.java]: Here could be used another locator

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    WebDriver driver = null;
+    // 1. OPEN TEST SITE BY URL
+    @BeforeMethod
+    public void setUp() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.get("https://epam.github.io/JDI/");
+
+    }
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: I think that this is looks like code duplication for the Ex1Tests

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    @Test
+    public void ex2Test() {
+        // 2. ASSERT BROWSER TITLE
+        String browserTitle = driver.getTitle();
+        assertEquals(browserTitle, "Home Page");
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Why do you use here `Hard Assert`?

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        //3. PERFORM LOGIN
+        driver.findElement(By.id("user-icon")).click();
+        driver.findElement(By.id("name")).sendKeys("epam");
+        driver.findElement(By.id("password")).sendKeys("1234");
+        driver.findElement(By.id("login-button")).click();
+
+
+        //4. ASSERT USER NAME IN THE LEFT-TOP SIDE OF SCREEN THAT USER IS LOGGINED
+        String userName = driver.findElement(By.id("user-name")).getText();
+        assertEquals(userName, "PITER CHAILOVSKII");
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Why do you use here `Hard Assert`?

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+        //6. CLICK ON SERVICE SUBCATEGORY IN THE LEFT SECTION AND CHECK THAT DROP DOWN CONTAINS OPTIONS
+        WebElement serviceInLeftSection = driver.findElement(By.xpath("//span[text() = 'Service']"));
+        serviceInLeftSection.click();
+        By locatorOfLeftSection = By.xpath("//ul[@class='sub']/li");
+        assertTrue(optionsInServiceAreAsExpected(locatorOfLeftSection));
+
+
+        //7. OPEN THROUGH THE HEADER MENU SERVICE -> DIFFERENT ELEMENTS PAGE
+        WebElement service = driver.findElement(By.xpath("//a[contains(text(),'Service')]"));
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Could be used **DOM locator**

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        service.click();
+        WebElement differentElements = service.findElement(By.xpath("//a[contains(text(),'Different elements')]"));
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Could be used *DOM Locator*

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+        //12. ASSERT THAT FOR EACH CHECKBOX THERE IS AN INDIVIDUAL LOG ROW AND VALUE IS CORRESPONDED TO THE STATUS OF CHECKBOX.
+        //18. ASSERT THAT FOR EACH CHECKBOX THERE IS AN INDIVIDUAL LOG ROW AND VALUE IS CORRESPONDED TO THE STATUS OF CHECKBOX.
+        for (WebElement iterCheckbox : checkboxes) {
+            iterCheckbox.click();
+            String iterCheckboxTitle = getParent(iterCheckbox).getText();
+            boolean statusOfCheckboxInLog = getLogRow().contains("true");
+            boolean statusOfCheckboxAfterClick = iterCheckbox.isSelected();
+            softAssert.assertTrue(getLogRow().contains(iterCheckboxTitle));
+            softAssert.assertEquals(statusOfCheckboxAfterClick, statusOfCheckboxInLog);
+        }
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: What are you trying to do by this loop?
It is looks like you have got miss understanding of the test case

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+        assertTrue(radioSelen.isSelected());
+
+
+        //14. ASSERT THAT FOR RADIOBUTTON THERE IS A LOG ROW AND VALUE IS CORRESPONDED TO THE STATUS OF RADIOBUTTON.
+        for (WebElement iterRadio : radios) {
+            iterRadio.click();
+            String iterRadioTitle = getParent(iterRadio).getText();
+            softAssert.assertTrue(getLogRow().contains(iterRadioTitle));
+
+        }
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: same story as for 12 and 18 steps

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+        assertTrue(click(checkboxWater).isSelected());
+        assertTrue(click(checkboxWind).isSelected());
+
+        // UNSELECT CHECKBOXES AND ASSERT
+        assertFalse(click(checkboxWater).isSelected());
+        assertFalse(click(checkboxWind).isSelected());
+
+
+        //12. ASSERT THAT FOR EACH CHECKBOX THERE IS AN INDIVIDUAL LOG ROW AND VALUE IS CORRESPONDED TO THE STATUS OF CHECKBOX.
+        //18. ASSERT THAT FOR EACH CHECKBOX THERE IS AN INDIVIDUAL LOG ROW AND VALUE IS CORRESPONDED TO THE STATUS OF CHECKBOX.
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Why here are 12 & 18 steps? Is this code work for the both?

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+        select.selectByVisibleText("Yellow");
+        assertTrue(dropdownOptions.get(3).isSelected());
+
+
+        // 16. ASSERT THAT FOR DROPDOWN THERE IS A LOG ROW AND VALUE IS CORRESPONDED TO THE SELECTED VALUE.
+        for (WebElement dropDownOption : dropdownOptions) {
+            dropDownOption.click();
+            String optionValue = dropDownOption.getAttribute("value");
+            softAssert.assertTrue(getLogRow().contains(optionValue));
+        }
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Same as for 12, 18, 14 steps

File: src/test/java/hw2/ex2/Ex2Tests.java [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+        softAssert.assertAll();
+    }
+
+
+    //17. CLOSE BROWSER
+    @AfterMethod
+    public void tearDown() {
+        driver.close();
```
Comment (khda91) [inline src/test/java/hw2/ex2/Ex2Tests.java]: Looks like code duplicatons for the `Ex1Tests`

File: src/test/java/hw1/dataproviders/DivideDataProvider.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
+package hw1.dataproviders;
```
Comment (khda91) [inline src/test/java/hw1/dataproviders/DivideDataProvider.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/dataproviders/MultiplyDataProvider.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
+package hw1.dataproviders;
```
Comment (khda91) [inline src/test/java/hw1/dataproviders/MultiplyDataProvider.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/dataproviders/SubtractDataProvider.java [hunk 16]
```diff
@@ -0,0 +1,26 @@
+package hw1.dataproviders;
```
Comment (khda91) [inline src/test/java/hw1/dataproviders/SubtractDataProvider.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/dataproviders/SumDataProvider.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
+package hw1.dataproviders;
```
Comment (khda91) [inline src/test/java/hw1/dataproviders/SumDataProvider.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/testclasses/DivideTest.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
+package hw1.testclasses;
```
Comment (khda91) [inline src/test/java/hw1/testclasses/DivideTest.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/testclasses/MultiplyTest.java [hunk 19]
```diff
@@ -0,0 +1,29 @@
+package hw1.testclasses;
```
Comment (khda91) [inline src/test/java/hw1/testclasses/MultiplyTest.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw1/testclasses/SubtractTest.java [hunk 20]
```diff
@@ -0,0 +1,30 @@
+package hw1.testclasses;
```
Comment (khda91) [inline src/test/java/hw1/testclasses/SubtractTest.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: src/test/java/hw2/ex1/Ex1Tests.java [hunk 21]
```diff
@@ -0,0 +1,140 @@
+package hw2.ex1;
```
Comment (khda91) [inline src/test/java/hw2/ex1/Ex1Tests.java]: Why hw1 is in the HW2 branch?
Why I could see this file as newly added?

File: pom.xml (hunk 1 above)
Comment (IliaRusecki) [reply pom.xml]: Ok, changed it

File: src/test/java/hw2/ex1/Ex1Tests.java (hunk 2 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex1/Ex1Tests.java]: Didn't understand quite well the question - is it a bad idea to loop through the header items and check them, or is it a bad locator?! but, anyway I changed the code and made separate objects for each header item.

File: src/test/java/hw2/ex1/Ex1Tests.java (hunk 3 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex1/Ex1Tests.java]: corrected

File: src/test/java/hw2/ex2/Ex2Tests.java (hunk 4 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex2/Ex2Tests.java]: Yeah, I made a common Base class, where I put this method

File: src/test/java/hw2/ex2/Ex2Tests.java (hunk 5 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex2/Ex2Tests.java]: sorry! changed it

File: src/test/java/hw2/ex2/Ex2Tests.java (hunk 6 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex2/Ex2Tests.java]: the same thing

File: src/test/java/hw2/ex2/Ex2Tests.java (hunk 7 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex2/Ex2Tests.java]: Yeah! with your help I explored method linkText :)

File: src/test/java/hw2/ex2/Ex2Tests.java (hunk 9 above)
Comment (IliaRusecki) [reply src/test/java/hw2/ex2/Ex2Tests.java]: Yes, think I didn't understand quite well all steps from 12 toll 18. and made some unnecessary work

**Pull request merged**